### PR TITLE
Matrics

### PR DESCRIPTION
**Pull request by pamod**

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/matrics/DataAccessMatrixManager.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * Will be responsible in managing between the carbon matrix manager
+ * Includes adding of timers and guages
+ */
+public class DataAccessMatrixManager {
+
+
+    //Holds the timers which will periodically trigger the matrix info
+    //Key - Timer definition, Value - timer object which will record info against the defined operation
+    //Map will be concurrently accessed during multiple db operations during run time.
+    private static Map<String, Timer> timers = new ConcurrentHashMap<String, Timer>();
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/matrics/DataAccessMatrixManager.java]: Make these /** */ style comments?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/cassandra/HectorDataAccessHelper.java [hunk 2]
```diff
@@ -839,6 +858,107 @@ public static void deleteIntegerRowListFromColumnFamily(String columnFamily, Lis
...
+
+        //Holds the messages that will be sent back to the API for delivery
+        Map<Long, List<AndesMessagePart>> messageContentBatch = new HashMap<Long, List<AndesMessagePart>>();
+
+        try {
+            //Specify the range of offsets in a message
+            Integer startOffSet = 0;
+            Integer maxOffset = Integer.MAX_VALUE;
+
+            //TODO remove this once the mid is taken out from hector
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/cassandra/HectorDataAccessHelper.java]: @hastef88 Could you check on this ?

File: modules/andes-core/broker/src/test/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImplTest.java [hunk 3]
```diff
@@ -111,6 +94,14 @@ public void testStoreRetrieveMessagePart() throws Exception {
             Assert.assertEquals(msgPart.getOffSet(), p.getOffSet());
         }
     }
+    
+    @Test
+    public void testGetMessageContent() {
+
```
Comment (ramith) [inline modules/andes-core/broker/src/test/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImplTest.java]: do we need this?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/matrics/DataAccessMatrixManager.java (hunk 1 above)
Comment (pamod) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/matrics/DataAccessMatrixManager.java]: +1 fixed with #54

File: modules/andes-core/broker/src/test/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImplTest.java (hunk 3 above)
Comment (pamod) [inline modules/andes-core/broker/src/test/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImplTest.java]: @Asitha could you check on this ?

Comment (Asitha) [inline modules/andes-core/broker/src/test/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImplTest.java]: I will update with the proper test case.

**Pull request merged**